### PR TITLE
Updated XSCreator.py for PHISICS interface 

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
#611
##### What are the significant changes in functionality due to this change request?
Acceleration of parsing input

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: framework/CodeInterfaces/PHISICS/XSCreator.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def fileReconstruction(self,deconstructedDict):
       Converts the formatted dictionary -> {'XS|FUEL1|U235|FISSION|1':1.30, 'XS|FUEL2|U238|ABS|2':4.69}
       into a dictionary of dictionaries that has the format -> {'XS':{'FUEL1':{'U235':{'FISSION':{'1':1.30}}}}, 'FUEL2':{'U238':{'ABS':{'2':4.69}}}}
       @ In, deconstructedDict, dictionary, dictionary of perturbed variables
-      @ Out, leanReconstructedDict, dict, nested dictionary of perturbed variables
+      @ Out, leanReconstructedDict, dictionary, nested dictionary of perturbed variables
```
Comment (alfoa) [inline framework/CodeInterfaces/PHISICS/XSCreator.py]: dict

File: framework/CodeInterfaces/PHISICS/XSCreator.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def formatXS(self,reaction):
     else:
       raise IOError('the type of cross section '+reaction+' cannot be processed. Refer to manual for available reactions.')
     return reactionTemplated
-
+
```
Comment (alfoa) [inline framework/CodeInterfaces/PHISICS/XSCreator.py]: remove trailing spaces...

run the script ```./developer_tools/delete_trailing_whitespace.sh framework```

File: framework/CodeInterfaces/PHISICS/XSCreator.py [hunk 3]
```diff
@@ -174,23 +175,12 @@ def fileReconstruction(self,deconstructedDict):
...
+    
+    myReconstructedDict = lambda: defaultdict(myReconstructedDict)
+    reconstructedDict = myReconstructedDict()
     for typeKey, value in deconstructedDict.iteritems():
       if typeKey in pertDictSet:
         keyWords = typeKey.split('|')
         reconstructedDict[keyWords[0]][keyWords[1]][keyWords[2]][keyWords[3]][keyWords[4]][keyWords[5]][keyWords[6]] = value
     leanReconstructedDict = self.cleanEmpty(reconstructedDict)
-    return leanReconstructedDict
+    return leanReconstructedDict
```
Comment (alfoa) [inline framework/CodeInterfaces/PHISICS/XSCreator.py]: add empty line

**Comment by alfoa** [comment]
@wangcj05 Since i added some modifications, I can not review this PR.
Can you do that?
Andrea

**Comment by wangcj05** [comment]
Yes, I can take a look.

________________________________

**Comment by aalfonsi** [comment]
yes lets close it.

**Pull request closed without merging**